components/sst_importer/src/import_file.rs

```diff
@@ -281,14 +285,77 @@ impl ImportDir {
         Ok(meta_info)
     }
 
+    /// check if api version of sst files are compatible
+    pub fn check_api_version(
+        &self,
+        metas: &[SstMeta],
+        key_manager: Option<Arc<DataKeyManager>>,
+        api_version: ApiVersion,
+    ) -> Result<bool> {
+        for meta in metas {
+            match (api_version, meta.api_version) {
+                (cur_version, meta_version) if cur_version == meta_version => continue,
+                // sometimes client do not know whether ttl is enabled, so a general V1 is accepted as V1ttl
+                (ApiVersion::V1ttl, ApiVersion::V1) => continue,
+                // import V1ttl as V1 will immediatly be rejected because it is never correct.
+                (ApiVersion::V1, ApiVersion::V1ttl) => return Ok(false),
+                // otherwise we are upgrade/downgrade between V1 and V2
+                // this can be done if all keys are written by TiDB
+                _ => {
+                    let path = self.join(meta)?;
+                    let path_str = path.save.to_str().unwrap();
+                    let env = get_env(key_manager.clone(), get_io_rate_limiter())?;
+                    let sst_reader = RocksSstReader::open_with_env(path_str, Some(env))?;
+
+                    for &(start, end) in TIDB_RANGES_COMPLEMENT {
+                        let mut unexpected_data_key = None;
+                        sst_reader.scan(start, end, false, |key, _| {
+                            unexpected_data_key = Some(key[DATA_KEY_PREFIX_LEN..].to_vec());
```

Comment:
Why needs `DATA_KEY_PREFIX_LEN` here?

Reply:
It's because we don't want the `z` prefix to show up in the user key when printing the error to user.

Reply:
I looked into `TIDB_RANGES_COMPLEMENT`, and it doesn't have `z` prefix.
So it means that, as arguments, keys won't contain the prefix, but the iterator will meet the prefix?

Reply:
`TIDB_RANGES_COMPLEMENT` do contains keys having `z` prefix. It have range `(&[b't' + 1], &[])`, which means [`'u'`, +∞).

Reply:
OK. The `[DATA_KEY_PREFIX_LEN..]` have been removed.